.github/workflows/tests.yml

```
@@ -29,20 +29,29 @@
       - name: Run database tests
         run: |
-          go test ./database/... -coverprofile=coverage.out
-          go tool cover -func=coverage.out | tail -n 1
+          go test ./database/... -coverprofile=coverage-database.out
+          go tool cover -func=coverage-database.out | tail -n 1
+
+      - name: Merge coverage reports
+        run: go tool cover -merge=coverage-*.out -o coverage.out
+
+      - name: Upload coverage artifact
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.7)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)